sourcegraph/cody
# refactor: move vscode context out of shared

refactor: move `vscode-context` from `lib/shared` to `vscode/editor-context`

Right now, `vscode-context.ts` is living in the share library does not work for other editors, and prevents other editors from implementing custom commands through the custom-prompt recipe.

This change allows other editors to implement editor-level context through the new `EditorContext class`, and enable other recipes to get editor context easily without needing to implement the same methods for each recipe.

Renamed '@sourcegraph/cody-shared/src/chat/prompts' to '@sourcegraph/cody-shared/src/chat/commands'

No current functions or features were altered in this PR.

Will clean up the `vscode/editor-context` file in a separate PR since that would involve updating the current functions.

## Test plan



1. build from this branch
2. try different commands
3. everything should work the same
